clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -925,7 +925,7 @@ public Map<TopicPartition, OffsetAndMetadata> fetchCommittedOffsets(final Set<To
 
         final Generation generationForOffsetRequest = generationIfStable();
         if (pendingCommittedOffsetRequest != null &&
-            !pendingCommittedOffsetRequest.sameRequest(partitions, generationForOffsetRequest)) {
+            !pendingCommittedOffsetRequest.sameOrSubsetRequest(partitions, generationForOffsetRequest)) {
```

Comment:
If we allow reusing the previous commit when matching subsets, the returned `Map<TopicPartition, OffsetAndMetadata>` will contain unrelated topic partitions. Therefore, should we adjust the return map of `consumer#committed` to match the input?

Reply:
Thanks for your remainder, updated.